homeassistant/components/wallbox/__init__.py

```diff
@@ -41,31 +44,29 @@ def __init__(self, station, wallbox, hass):
             update_interval=timedelta(seconds=UPDATE_INTERVAL),
         )
 
-    def _authenticate(self):
+    def _authenticate(self) -> None:
         """Authenticate using Wallbox API."""
         try:
             self._wallbox.authenticate()
-            return True
         except requests.exceptions.HTTPError as wallbox_connection_error:
             if wallbox_connection_error.response.status_code == HTTPStatus.FORBIDDEN:
                 raise InvalidAuth from wallbox_connection_error
             raise ConnectionError from wallbox_connection_error
 
-    def _validate(self):
+    def _validate(self) -> None:
         """Authenticate using Wallbox API."""
         try:
             self._wallbox.authenticate()
-            return True
         except requests.exceptions.HTTPError as wallbox_connection_error:
             if wallbox_connection_error.response.status_code == 403:
                 raise InvalidAuth from wallbox_connection_error
             raise ConnectionError from wallbox_connection_error
 
-    def _get_data(self):
+    def _get_data(self) -> dict[str, Any]:
```

Comment:
@hesselonline If you are interested, this could be improved even more by using a `TypedDict` as return value.
I haven't done that yet as I don't know what exactly `wallbox.getChargerStatus` will return. Couldn't find an API documentation for it and don't use that integration personally.

Would you be able to print the `data` dict and post it here?

--
That's purely optional at this point though, so only if you like.

Reply:
yes, I can do that alter this week when giving it a test run. Small concern, what happens when the upstream pypi packages adds/removes a key? Won't this make it more prone to breaking / TypeErrors?

Reply:
> Small concern, what happens when the upstream pypi packages adds/removes a key? Won't this make it more prone to breaking / TypeErrors?

Adding a key doesn't matter. If they remove one, it would break anyway. That's not really something we can handle that well as the pypi package just returns the API json response anyway.
It would be nice to use a `TypedDict`, but honestly we could also leave it just as it is now. That's perfectly fine, too!

Reply:
This is a complete Data dump (removed sensitive data):
`{
    "user_id":1,
    "user_name":"default",
    "car_id":1,
    "car_plate":"",
    "depot_price":0.3,
    "last_sync":"2021-11-13 07:47:48",
    "power_sharing_status":256,
    "mid_status":0,
    "status_id":181,
    "status_description":"Connected: waiting for car demand",
    "name":"Copper Business SN XXXXXXX",
    "charging_power":0,
    "max_available_power":25,
    "depot_name":"Family",
    "charging_speed":0,
    "added_range":405,
    "added_energy":48.672,
    "added_green_energy":0,
    "charging_time":82829,
    "cost":0,
    "current_mode":3,
    "state_of_charge":"None",
    "ocpp_status":1,
    "config_data":{
       "charger_id":XXXXXX,
       "uid":"XXXXXXYYYYYYYZZZZZZ",
       "serial_number":"XXXXX",
       "name":"Copper Business SN XXXXX",
       "locked":0,
       "auto_lock":0,
       "auto_lock_time":60,
       "multiuser":0,
       "max_charging_current":21,
       "language":"EN",
       "icp_max_current":25,
       "grid_type":1,
       "energy_price":0.3,
       "energyCost":{
          "value":0.3,
          "inheritedGroupId":"None"
       },
       "unlock_user_id":"None",
       "power_sharing_config":256,
       "purchased_power":0,
       "show_name":1,
       "show_lastname":1,
       "show_email":1,
       "show_profile":1,
       "show_default_user":1,
       "gesture_status":7,
       "home_sharing":1,
       "dca_status":0,
       "connection_type":0,
       "max_available_current":25,
       "live_refresh_time":30,
       "update_refresh_time":300,
       "owner_id":1234567,
       "remote_action":0,
       "rfid_type":"None",
       "charger_has_image":0,
       "sha256_charger_image":"None",
       "plan":{
          "plan_name":"Basic",
          "features":[
             "DEFAULT_FEATURE",
             "POWER_BOOST",
             "MOBILE_CONNECTIVITY",
             "STATISTICS",
             "AUTOMATIC_REPORTING"
          ]
       },
       "sync_timestamp":1636787853,
       "currency":{
          "id":1,
          "name":"Euro Member Countries",
          "symbol":"€",
          "code":"EUR"
       },
       "charger_load_type":"Private",
       "contract_charging_available":false,
       "country":{
          "id":166,
          "code":"NLD",
          "iso2":"NL",
          "name":"HOLANDA",
          "phone_code":"31"
       },
       "state":"None",
       "part_number":"CPBX-X-5-X",
       "software":{
          "updateAvailable":false,
          "currentVersion":"5.3.16",
          "latestVersion":"5.3.16"
       },
       "available":1,
       "operation_mode":"wallbox",
       "ocpp_ready":"ocpp_1.6j",
       "tariffs":[
          
       ],
       "mid_enabled":0,
       "mid_margin":1,
       "mid_margin_unit":1,
       "mid_serial_number":"",
       "mid_status":0,
       "session_segment_length":0,
       "group_id":91900
    },
    "max_charging_current":21
 }`

Reply:
We only use the keys that are in SENSOR_TYPES dict BTW

Reply:
Thanks a lot! That helps. I would suggest to do the `TypedDict` in a separate PR after this one is merged. That way we could more easily decide if it's helpful at all or close it if not.

--
Could you post a dump from when a car is charging? Just to see how the values are changing. (The `config_data` key doesn't matter too much.)

Reply:
Sure, have to wait until the car is empty-ish again ;-). Structure of the dict remains the same however, mainly the status that changes